src/Silicon.cpp

```diff
@@ -874,124 +1135,493 @@ namespace galsim {
         Bounds<int> b = target.getBounds();
         double addedFlux = 0.;
 
+	// Get everything out of C++ classes and into arrays/structures suitable for GPU
+	// photons
+        // can't get pointers to internal data from a const reference
+        PhotonArray& photonsMutable = const_cast<PhotonArray&>(photons);        
+	double* photonsX = photonsMutable.getXArray();
+	double* photonsY = photonsMutable.getYArray();
+	double* photonsDXDZ = photonsMutable.getDXDZArray();
+	double* photonsDYDZ = photonsMutable.getDYDZArray();
+	double* photonsFlux = photonsMutable.getFluxArray();
+	double* photonsWavelength = photonsMutable.getWavelengthArray();
```

Comment:
Do you really need pointers to mutable values?  If not, you should just add const versions of these in the PhotonArray class that returns `const double*` pointers rather than using const_cast.